components/brave_wallet_ui/page/screens/swap/utils.ts

```diff
@@ -0,0 +1,193 @@
+}
+
+export function getZeroExToAmount({
+  quote,
+  toToken
+}: {
+  quote: BraveWallet.ZeroExQuote
+  toToken: BraveWallet.BlockchainToken
+}): Amount {
+  return new Amount(quote.buyAmount).divideByDecimals(toToken.decimals)
+}
+
+export function getZeroExQuoteOptions({
+  quote,
+  fromNetwork,
+  fromToken,
+  toToken,
+  spotPrices,
+  defaultFiatCurrency
+}: {
+  quote: BraveWallet.ZeroExQuote
+  fromNetwork: BraveWallet.NetworkInfo
+  fromToken: BraveWallet.BlockchainToken
+  toToken: BraveWallet.BlockchainToken
+  spotPrices: SpotPriceRegistry
+  defaultFiatCurrency: string
+}): QuoteOption[] {
+  const networkFee = getZeroExNetworkFee({ quote, fromNetwork })
+
+  return [
+    {
+      label: '',
+      fromAmount: new Amount(quote.sellAmount).divideByDecimals(
+        fromToken.decimals
+      ),
+      toAmount: getZeroExToAmount({ quote, toToken }),
+      minimumToAmount: undefined,
+      fromToken,
+      toToken,
+      rate: new Amount(quote.buyAmount)
+        .divideByDecimals(toToken.decimals)
+        .div(new Amount(quote.sellAmount).divideByDecimals(fromToken.decimals)),
+      impact: new Amount(quote.estimatedPriceImpact),
+      sources: quote.sources
+        .map((source) => ({
+          name: source.name,
+          proportion: new Amount(source.proportion)
+        }))
+        .filter((source) => source.proportion.gt(0)),
+      routing: 'split', // 0x supports split routing only
+      networkFee,
+      networkFeeFiat: networkFee.isUndefined()
+        ? ''
+        : networkFee
+            .times(
+              getTokenPriceAmountFromRegistry(
+                spotPrices,
+                makeNetworkAsset(fromNetwork)
+              )
+            )
+            .formatAsFiat(defaultFiatCurrency)
+    }
+  ]
+}
+
+function getJupiterNetworkFee({
+  quote,
+  fromNetwork
+}: {
+  quote: BraveWallet.JupiterQuote
+  fromNetwork: BraveWallet.NetworkInfo
+}): Amount {
+  if (!fromNetwork) {
+    return Amount.empty()
+  }
+
+  return new Amount('0.000005')
+}
+
+export function getJupiterFromAmount({
+  quote,
+  fromToken
+}: {
+  quote: BraveWallet.JupiterQuote
+  fromToken: BraveWallet.BlockchainToken
+}): Amount {
+  return new Amount(quote.inAmount).divideByDecimals(fromToken.decimals)
+}
+
+export function getJupiterToAmount({
+  quote,
+  toToken
+}: {
+  quote: BraveWallet.JupiterQuote
+  toToken: BraveWallet.BlockchainToken
+}): Amount {
+  return new Amount(quote.outAmount).divideByDecimals(toToken.decimals)
+}
+
+export function getJupiterQuoteOptions({
+  quote,
+  fromNetwork,
+  fromToken,
+  toToken,
+  spotPrices,
+  defaultFiatCurrency
+}: {
+  quote: BraveWallet.JupiterQuote
+  fromNetwork: BraveWallet.NetworkInfo
+  fromToken: BraveWallet.BlockchainToken
+  toToken: BraveWallet.BlockchainToken
+  spotPrices: SpotPriceRegistry
+  defaultFiatCurrency: string
+}): QuoteOption[] {
+  const networkFee = getJupiterNetworkFee({ quote, fromNetwork })
+
+  return [
+    {
+      label: '',
```

Comment:
Can we comment on why the label is empty?

Reply:
https://github.com/brave/brave-core/pull/22285/commits/fcb1bcdb97390a71f10b71c5f98cc022da900914